### PR TITLE
Fix test data

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Fix test data according to the transition to not seeding test data into environments and at the same time using the new shared actor register.

All SQL scripts have been revamped.
image

## References

* #924

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    ) ON [PRIMARY]
+    GO
+
+CREATE TABLE [MessageHub].[AvailableChargeLinksData](
+    [Id] [uniqueidentifier] NOT NULL,
+    [RecipientId] [varchar](70) NOT NULL,
+    [RecipientRole] [int] NOT NULL,
+    [BusinessReasonCode] [int] NOT NULL,
+    [ChargeId] [varchar](70) NOT NULL,
+    [ChargeOwner] [varchar](70) NOT NULL,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql:185]: The type and length of `RecipientId`, `ChargeId` and `ChargeOwner` deviates from `[MessageHub].[AvailableChargeData]`. I think we should align with `[MessageHub].[AvailableChargeData]`

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    [ChargeType] [int] NOT NULL,
+    [MeteringPointId] [varchar](70) NOT NULL,
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql:187]: Suggested to align with metering point table: 
```suggestion
    [MeteringPointId] [nvarchar](50) NOT NULL,
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql (hunk 1 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202201200800 Create tables.sql:185]: Ok. Fixed - including metering point id.
Nice catch by the way.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200830 Add market participants.sql [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8510000000006', 1, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8510000000013', 1, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8510000000020', 1, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8510000000037', 1, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8100000000016', 2, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8100000000023', 2, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8100000000030', 2, 1);
+INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'8900000000005', 2, 0);
+-- System operator is seeded as it is required in order to be able to seed default charges
+--INSERT INTO [Charges].[MarketParticipant] VALUES (NEWID(),'5790000432752', 3, 1);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200830 Add market participants.sql:13]: nit:
```suggestion
-- System operator has already been seeded in script 'Add default charges owner'
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200831 Add metering points.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO [Charges].[MeteringPoint] VALUES (NEWID(), '571313180000000005', 1, 800, '2020-12-31T23:00:00', 2, 3)
+INSERT INTO [Charges].[MeteringPoint] VALUES (NEWID(), '701313180000000005', 3, 800, '2020-12-31T23:00:00', 1, 3)
+INSERT INTO [Charges].[MeteringPoint] VALUES (NEWID(), '901313180000000005', 1, 800, '2020-12-31T23:00:00', 1, 3)
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200831 Add metering points.sql:3]: Right now the values doesn't pose a problem, but they probably will later, when we begin modus checking them.
I suggest to do this change now, to make them modus 11 valid. 
And Please update CreateDefaultChargeLinksReceiverTests.cs where they are used.
```suggestion
INSERT INTO [Charges].[MeteringPoint] VALUES (NEWID(), '571313180000000012', 3, 800, '2020-12-31T23:00:00', 1, 3)
INSERT INTO [Charges].[MeteringPoint] VALUES (NEWID(), '901313180000000029', 1, 800, '2020-12-31T23:00:00', 1, 3)
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200832 Add charges and links.sql [hunk 5]
```diff
@@ -11,10 +11,10 @@ SELECT @chargeId2 = NEWID()
...
-INSERT INTO charges.charge VALUES (@chargeId3, 'TestFee', 2, @chargeOwnerId, 0, 3, 0, 'Description...', 'Grid Access Provider test fee', 2, '1999-12-31 23:00:00', NULL)
-INSERT INTO charges.charge VALUES (@chargeId4, 'TestSubscription', 1, @chargeOwnerId, 0, 4, 0, 'Description...', 'Grid Access Provider test subscription', 2, '1999-12-31 23:00:00', NULL)
+INSERT INTO charges.charge VALUES (@chargeId, 'TestTariff', 3, @chargeOwnerId, 0, 2, 0, 'Description...', 'Grid Access Provider test tariff', 1, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
+INSERT INTO charges.charge VALUES (@chargeId2, 'TestTariff2', 3, @chargeOwnerId, 0, 2, 1, 'Description...', 'Grid Access Provider test tariff2', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
+INSERT INTO charges.charge VALUES (@chargeId3, 'TestFee', 2, @chargeOwnerId, 0, 3, 0, 'Description...', 'Grid Access Provider test fee', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
+INSERT INTO charges.charge VALUES (@chargeId4, 'TestSubscription', 1, @chargeOwnerId, 0, 4, 0, 'Description...', 'Grid Access Provider test subscription', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200832 Add charges and links.sql:17]: First, good catch with the missing end dates 👍 
I made a mistake when I initially added these new charges, as some of them violate the allowed name length(10).
I suggest we change them like this. I checked the code base, the names aren't used in any tests, so this should be safe.
```suggestion
INSERT INTO charges.charge VALUES (@chargeId, 'TestTariff', 3, @chargeOwnerId, 0, 2, 0, 'Description...', 'Grid Access Provider test tariff', 1, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
INSERT INTO charges.charge VALUES (@chargeId2, 'TestTar002', 3, @chargeOwnerId, 0, 2, 1, 'Description...', 'Grid Access Provider test tariff2', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
INSERT INTO charges.charge VALUES (@chargeId3, 'TestFee', 2, @chargeOwnerId, 0, 3, 0, 'Description...', 'Grid Access Provider test fee', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
INSERT INTO charges.charge VALUES (@chargeId4, 'TestSub', 1, @chargeOwnerId, 0, 4, 0, 'Description...', 'Grid Access Provider test subscription', 2, '1999-12-31 23:00:00', '9999-12-31 23:59:59')
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200831 Add metering points.sql (hunk 4 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Test/202201200831 Add metering points.sql]: I don't think we should ever validate these identification numbers as they are owned by the actor register. We should simply take them as-is from there. But I still agree that it's prettier to use valid numbers.

**Review by prtandrup — APPROVED** [review]
LGTM, good job 👍

**Pull request merged**